src/Symfony/Component/Config/Definition/PrototypedArrayNode.php

```diff
@@ -304,9 +304,10 @@ protected function mergeValues($leftSide, $rightSide)
             return $rightSide;
         }
 
+        $isAssoc = array_keys($rightSide) !== range(0, \count($rightSide) - 1);
```

Comment:
What's about using `array_is_list` and requiring the `php81` polyfill?

Reply:
Updated PR with this polyfill.